jbehave-support-core/src/main/java/org/jbehavesupport/core/rest/RestServiceHandler.java

```diff
@@ -461,12 +461,38 @@ public void verifyResponse(String expectedStatus, ExamplesTable data) {
         if (!StringUtils.isEmpty(expectedStatus)) {
             verifyResponseStatus(actualResponseStatus, parseHttpStatus(expectedStatus), actualResponseMessage);
         }
-        if (ExamplesTableUtil.tableContains(data, NAME, STATUS_HEADER::equals)) {
-            verifyResponseStatus(actualResponseStatus, parseHttpStatus(getValue(data, NAME, STATUS_HEADER, EXPECTED_VALUE)), actualResponseMessage);
+        if (ExamplesTableUtil.tableContains(convertedData, NAME, STATUS_HEADER::equals)) {
+            verifyResponseStatus(actualResponseStatus, parseHttpStatus(getValue(convertedData, NAME, STATUS_HEADER, EXPECTED_VALUE)), actualResponseMessage);
         }
 
-        verifyResponseHeaders(actualHeaders, data, actualResponseMessage);
-        verifyResponseJson(actualResponseBody, data, actualResponseMessage);
+        verifyResponseHeaders(actualHeaders, convertedData, actualResponseMessage);
+        verifyResponseJson(actualResponseBody, convertedData, actualResponseMessage);
+    }
+
+    private ExamplesTable convertCollectionNotation(ExamplesTable data) {
+        if (data != null && !data.getRows().isEmpty()) {
+            List<Map<String, String>> newMapList = data.getRows().stream()
+                .map(map -> convertIfNeeded(map))
+                .collect(Collectors.toList());
+            return data.withRows(newMapList);
+        } else {
+            return data;
+        }
+    }
+
+    private Map<String, String> convertIfNeeded(Map<String, String> map) {
+        String name = map.get(ExampleTableConstraints.NAME);
+        Matcher mainMatcher = Pattern.compile(PERIOD_REGEX).matcher(name);
```

Comment:
What entitles this matcher to be main?

Reply:
Nothing, it is a leftower from a different approach with multiple matchers I tried during development; will be renamed